app/controllers/devise_overrides/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,69 @@
+class DeviseOverrides::OmniauthCallbacksController < DeviseTokenAuth::OmniauthCallbacksController
+  include EmailHelper
+
+  def omniauth_success
+    get_resource_from_auth_hash
+
+    if @resource.nil?
+      return redirect_to login_page_url(error: 'no-account-found') unless account_signup_allowed?
+      return redirect_to login_page_url(error: 'business-account-only') unless validate_business_account?
+
+      create_account_for_user
+      token = @resource.send(:set_reset_password_token)
+      frontend_url = ENV.fetch('FRONTEND_URL', nil)
+      return redirect_to "#{frontend_url}/app/auth/password/edit?config=default&reset_password_token=#{token}"
+    end
+
+    @resource.skip_confirmation! if confirmable_enabled?
+
+    # once the resource is found and verified
+    # we can just send them to the login page again with the SSO params
+    # that will log them in
+    encoded_email = ERB::Util.url_encode(@resource.email)
+    redirect_to login_page_url(email: encoded_email, sso_auth_token: @resource.generate_sso_auth_token)
+  end
+
+  def login_page_url(error: nil, email: nil, sso_auth_token: nil)
+    frontend_url = ENV.fetch('FRONTEND_URL', nil)
+    params = { email: email, sso_auth_token: sso_auth_token }.compact
+    params[:error] = error if error.present?
+
+    "#{frontend_url}/app/login?#{params.to_query}"
+  end
+
+  def account_signup_allowed?
+    # set it to true by default, this is the behaviour across the app
+    ActiveModel::Type::Boolean.new.cast(ENV.fetch('ENABLE_ACCOUNT_SIGNUP', true))
```

Comment:
let's use `GlobalConfigService.load('ENABLE_ACCOUNT_SIGNUP', 'false')` instead. 
Chatwoot also allows configuration from UI for this flag. hence we will have to account for that.

Reply:
@sojan-official made the changes, PTAL